app/logger_config.py

```diff
@@ -12,15 +14,68 @@ def format(self, record):
         else:
             record.url = None
             record.remote_addr = None
-
         return super().format(record)
 
 
+class CloudWatchHandler(logging.Handler):
+    def __init__(self, log_group, stream_name):
+        super().__init__()
+        self.client = boto3.client("logs")
+        self.log_group = log_group
+        self.stream_name = stream_name
+
+    def emit(self, record):
+        try:
+            log_entry = self.format(record)
+            self.client.put_log_events(
+                logGroupName=self.log_group,
+                logStreamName=self.stream_name,
+                logEvents=[
+                    {
+                        "timestamp": int(record.created * 1000),
+                        "message": log_entry,
+                    }
+                ],
+            )
+        except ClientError as e:
+            print(f"Error sending log to CloudWatch: {e}")
+
+
 def setup_logging(app):
     formatter = RequestFormatter(
         "[%(asctime)s] %(remote_addr)s requested %(url)s\n"
         "%(levelname)s in %(module)s: %(message)s"
     )
-    default_handler.setFormatter(formatter)
 
-    app.logger.setLevel(logging.INFO)
+    if os.getenv("ENV") == "aws":
+        app_logger = logging.getLogger("app_logger")
+        app_logger.setLevel(logging.INFO)
+        app_handler = CloudWatchHandler(
+            log_group="app-logs", stream_name="app-log-stream"
+        )
+        app_handler.setFormatter(formatter)
+        app_logger.addHandler(app_handler)
+
+        audit_logger = logging.getLogger("audit_logger")
+        audit_logger.setLevel(logging.INFO)
+        audit_handler = CloudWatchHandler(
+            log_group="audit-logs", stream_name="audit-log-stream"
+        )
+        audit_handler.setFormatter(formatter)
+        audit_logger.addHandler(audit_handler)
+
+    else:
```

Comment:
can we move all the shared bits out of the if-else block to reduce duplication / simplify things. I think just the `audit_handler =` line needs to be conditional, right?